corehq/apps/enterprise/views.py

```diff
@@ -323,3 +329,65 @@ def refresh_item(self, item_id):
 
     def post(self, *args, **kwargs):
         return self.paginate_crud_response
+
+
+@require_can_edit_or_view_web_users
+@require_superuser
+def enterprise_permissions(request, domain):
+    account = BillingAccount.get_account_by_domain(domain)
+    all_domains = set(account.get_domains())
+    ignored_domains = set(account.permissions_ignore_domains)
+    controlled_domains = all_domains - ignored_domains - set([account.permissions_source_domain])
+
+    context = {
+        'domain': domain,
+        'all_domains': sorted(all_domains),
+        'source_domain': account.permissions_source_domain,
+        'ignored_domains': sorted(list(ignored_domains)),
+        'controlled_domains': sorted(list(controlled_domains)),
+        'current_page': {
+            'page_name': _('Enterprise Permissions'),
+            'title': _('Enterprise Permissions'),
+        }
+    }
+    return render(request, "enterprise/enterprise_permissions.html", context)
+
+
+@require_superuser
+@require_POST
+def toggle_enterprise_permission(request, domain, target_domain):
```

Comment:
Rather than toggling I think it it generally safer to pass in the desired value from the view to make sure that the change reflects the user's intent.

Reply:
a2848c0901f27f23c31e8cf47f7d9be57ca2de98